lib/semian/circuit_breaker.rb

```diff
@@ -43,70 +44,77 @@ def mark_failed(_error)
 
     def mark_success
       return unless half_open?
-      @successes += 1
+      @successes.increment
       close if success_threshold_reached?
     end
 
     def reset
-      @errors = []
-      @successes = 0
+      @errors.clear
+      @successes.value = 0
```

Comment:
The PR doesn't mention anything about `Integer::value=` though. Isn't that the issue here?